### PR TITLE
New documentation layout

### PR DESCRIPTION
**Pull request by jorgepiloto**
Solves for #975 by adding a new documentation layout.

**Old layout**            |  **New layout** 
:-------------------------:|:-------------------------:
old_layout  |  new_layout

Notice that some filenames were changed together with their heading sections just to match desired documentation structure :relaxed:

### CODE REVIEW

**Comment by astrojuanlu** [comment]
@jorgepiloto Can you please rebase? :)

File: docs/source/bibliography.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-References
-==========
+Bibliography
```
Comment (astrojuanlu) [inline docs/source/bibliography.rst:1]: Yay!

File: docs/source/user_guide.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
-User guide
-==========
+Use poliastro
```
Comment (astrojuanlu) [inline docs/source/user_guide.rst:1]: I think we can keep this as "user guide", it's a very common term

Comment (jorgepiloto) [reply docs/source/user_guide.rst:1]: Sure, no problem with that!

**Pull request merged**